romancal/outlier_detection/outlier_detection_step.py

```diff
@@ -1,37 +1,202 @@
+        step.do_detection()
 
-        # No reference files
-        # reference_file_model = {}
-        # Do the outlier detection
-        # output_model = outlier_detection.do_correction(
-        #     input_model,
-        #     reference_file_model,
-        # )
+        state = "COMPLETE"
+        if self.input_container:
+            if not self.save_intermediate_results:
+                self.log.debug(
+                    "The following files will be deleted since save_intermediate_results=False:"
+                )
+            for model in self.input_models:
+                model.meta.cal_step.outlier_detection = state
+                if not self.save_intermediate_results:
+                    #  Remove unwanted files
+                    crf_path = self.make_output_path(basepath=model.meta.filename)
+                    #  These lines to be used when/if outlier_i2d files follow output_dir
+                    #  crf_file = os.path.basename(crf_path)
+                    #  outlr_path = crf_path.replace(crf_file, outlr_file)
+                    outlr_file = model.meta.filename.replace("cal", "outlier_i2d")
+                    blot_path = crf_path.replace("crf", "blot")
+                    median_path = blot_path.replace("blot", "median")
 
-        output_model = input_model
+                    for fle in [outlr_file, blot_path, median_path]:
+                        if os.path.isfile(fle):
+                            os.remove(fle)
+                            self.log.debug(f"    {fle}")
+        else:
+            self.input_models.meta.cal_step.outlier_detection = state
+        return self.input_models
 
-        # Close the input and reference files
-        input_model.close()
+    def check_input(self):
+        """Use this method to determine whether input is valid or not."""
+        if self.input_container:
+            self._check_input_container()
+        else:
+            self._check_input_cube()
 
-        if self.save_results:
-            try:
-                self.suffix = "outlier_detection"
-            except AttributeError:
-                self["suffix"] = "outlier_detection"
+    def _check_input_container(self):
+        """Check to see whether input is the expected ModelContainer object."""
+        ninputs = len(self.input_models)
+        if not isinstance(self.input_models, ModelContainer):
+            self.log.warning("Input is not a ModelContainer")
+            self.log.warning("Outlier detection step will be skipped")
+            self.valid_input = False
+        elif ninputs < 2:
+            self.log.warning(f"Input only contains {ninputs} exposure")
+            self.log.warning("Outlier detection step will be skipped")
+            self.valid_input = False
+        else:
+            self.valid_input = True
+            self.log.info(f"Performing outlier detection on {ninputs} inputs")
 
-        return output_model
+    def _check_input_cube(self):
```

Comment:
Remove. We are not dealing with `CubeModel`s

Reply:
Done.